app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/UpdateLegacyCatalogInventoryAtStockSettingTime.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Plugin\CatalogInventory\Model\ResourceModel\Stock;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\CatalogInventory\Model\Stock\Item;
+use Magento\CatalogInventory\Model\ResourceModel\Stock\Item as ResourceItem;
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryApi\Api\Data\SourceItemInterfaceFactory;
+use Magento\InventoryApi\Api\SourceItemsSaveInterface;
+use Magento\InventoryCatalog\Model\DefaultSourceProvider;
+
+/**
+ * Class provides around Plugin on Magento\CatalogInventory\Model\ResourceModel\Stock\Item::save
+ * to update data in Inventory source item
+ */
+class UpdateLegacyCatalogInventoryAtStockSettingTime
```

Comment:
Why do we have this naming `UpdateLegacyCatalogInventoryAtStockSettingTime ` if the logic of this class is updating MSI SourceItem at the time when Legacy CatalogInventory Stock Item is updated.
So, we don't update  Legacy CatalogInventory Stock Item  (as plugin's name says), but doing opposite thing - we update new MSI Source Item